webhook/server.js

```diff
@@ -0,0 +1,377 @@
+            ...room.custom_data
+        })
+        await streamChannel.create()
+        return streamChannel
+    }
+
+    // converts a chatKit room to a stream channel
+    async toStreamMessage(streamChannel, chatKitMessage) {
+        const streamMessage = {
+            id: chatKitMessage.id.toString(),
+            user_id: chatKitMessage.user_id,
+            text: chatKitMessage.text,
+            attachments: [],
+        }
+
+        //build attachments
+        for (let i = 0; i < chatKitMessage.parts.length; i++) {
+            const part = chatKitMessage.parts[i]
+            if (part.url) {
+                this.addUrlPart(streamMessage, part)
+            } else if (part.attachment) {
+                    await this.addAttachmentPart(streamChannel, streamMessage, part)
+            } else {
+                //inline part
+                switch (part.type) {
+                    case 'text/plain':
+                        streamMessage.text = part.content;
+                }
+            }
+        }
+
+        return streamMessage
+    }
+
+    // url parts are added to the message body and scrapped server side
+    addUrlPart(streamMessage, part) {
+        if (streamMessage.text) {
+            streamMessage.text += "\n"
+            streamMessage.text += part.url
+        } else {
+            streamMessage.text = part.url
+        }
+    }
+
+    async addAttachmentPart(streamChannel, streamMessage, part) {
+        const isImage = part.type.startsWith("image/")
+        const streamAttachment = {
+            id: part.attachment.id,
+            file_size: part.attachment.size,
+            text: part.attachment.name,
+            mime_type: part.type,
+            type: part.type.split('/')[0],
+            ...part.attachment.custom_data,
+        }
+
+        let tmpObj = tmp.fileSync({mode: '0644', prefix: 'pusher-download-'});
+        const { buffer,err} = await this.downloadFile(part);
+        if (err){
+            throw err
+        }
+        fs.appendFileSync(tmpObj.fd, buffer)
+
+        if (isImage) {
+            const url = `${streamChannel._channelURL()}/image`
+            let resp = await this.sendFile(this.streamClient(), url, tmpObj, part.attachment.id, part.type, streamMessage.user_id)
+            resp.title_link = resp.file
+            resp.image_url = resp.file
+        } else {
+            const url = `${streamChannel._channelURL()}/file`
+            let resp = await this.sendFile(this.streamClient(), url, tmpObj, part.attachment.id, 'application/octet-stream', streamMessage.user_id)
+            resp.asset_url = resp.file
+        }
+        await tmpObj.removeCallback();
+        streamMessage.attachments.push(streamAttachment)
+    }
+
+    downloadFile(part) {
+        return new Promise(resolve => {
+            request(part.attachment.download_url, {encoding: null}, (err, response, buffer) => {
+                resolve({buffer,err});
+            })
+        })
+    }
```

Comment:
This won't work for any secured downloads, such as S3 or the chatkit file service ontop of S3

Reply:
Right, this works for the default setup.
we can't cover all possible service providers/custom upload logic.

You can ways modify this according to your needs, and we can adjust the import process on our side